hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -165,6 +165,15 @@ public void discoverTypes(String className, byte[] originalBytes) {
 	}
 
 	private DynamicType.Builder<?> doEnhance(Supplier<DynamicType.Builder<?>> builderSupplier, TypeDescription managedCtClass) {
+		// skip if the class was already enhanced. This is very common in WildFly as classloading is highly concurrent.
+		// We need to ensure that no class is instrumented multiple times as that might result in incorrect bytecode.
+		// N.B. there is a second check below using a different approach: checking for the marker interfaces; I'm currently
+		// inclined to keep both checks for better backwards compatibility.
```

Comment:
```suggestion
		// N.B. there is a second check below using a different approach: checking for the marker interfaces,
		// which does not address the case of extended bytecode enhancement
		// (because it enhances classes that do not end up with these marker interfaces).
		// I'm currently inclined to keep the other check anyway, for better backwards compatibility.
```

Reply:
Thanks, added this and elaborated a bit further. Also adapted the second logging message so that it's clearer which condition caused us to skip it.